pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/realtime_to_offline_segments/RealtimeToOfflineSegmentsTaskExecutor.java

```diff
@@ -132,67 +130,66 @@ public void preProcess(PinotTaskConfig pinotTaskConfig) {
     String offlineTableName = TableNameBuilder.OFFLINE.tableNameWithType(rawTableName);
     TableConfig tableConfig = getTableConfig(offlineTableName);
     Schema schema = getSchema(offlineTableName);
-    Set<String> schemaColumns = schema.getPhysicalColumnNames();
     String timeColumn = tableConfig.getValidationConfig().getTimeColumnName();
     DateTimeFieldSpec dateTimeFieldSpec = schema.getSpecForTimeColumn(timeColumn);
     Preconditions
         .checkState(dateTimeFieldSpec != null, "No valid spec found for time column: %s in schema for table: %s",
             timeColumn, offlineTableName);
+    SegmentProcessorConfig.Builder segmentProcessorConfigBuilder =
+        new SegmentProcessorConfig.Builder().setTableConfig(tableConfig).setSchema(schema);
 
-    long windowStartMs = Long.parseLong(configs.get(MinionConstants.RealtimeToOfflineSegmentsTask.WINDOW_START_MS_KEY));
-    long windowEndMs = Long.parseLong(configs.get(MinionConstants.RealtimeToOfflineSegmentsTask.WINDOW_END_MS_KEY));
+    long windowStartMs = Long.parseLong(configs.get(RealtimeToOfflineSegmentsTask.WINDOW_START_MS_KEY));
+    long windowEndMs = Long.parseLong(configs.get(RealtimeToOfflineSegmentsTask.WINDOW_END_MS_KEY));
     _nextWatermark = windowEndMs;
 
-    String timeColumnTransformFunction =
```

Comment:
Removing `TIME_COLUMN_TRANSFORM_FUNCTION` means that RealtimeToOffline task is not backward compatible after this change. 

Two options:
1. Add deprecated annotation & keep the backward compatibility instead of removing this
2. Remove this and include it in the release note

Reply:
Yes. Since rollup is not supported yet, decide to remove this config and mark it as backward-incompatible. The changes are listed within the release note.